index.js

```diff
@@ -99,6 +99,7 @@ function wrapFunction(config, req, res, cb) {
   // @TODO may want to put this behind a disable flag
   if (req.method === 'OPTIONS') {
     res.status(200).send();
+    return;
```

Comment:
Just out of curiosity: why does this help?

Reply:
Because send() method doesn't ends execution of current method.

Before this fix the flow looked like this:
1. request method === OPTIONS, then send status 200 (here it should end any other actions because we sent response)
2. try to parse and get token from request (which is not included in OPTIONS request)
3. token not found, send status 401... oops we try to set status again but it was sent before, we get error about `setting request headers after they are sent`

This worked for other HTTP methods because we leave out this `if` statement

With this fix we end functions execution after sending response (for OPTIONS method)

Reply:
Still don't fully get it. Do when method is OPTIONS, we just blindly send 200 back and return ? So why do we need this anyway? What's more: what about the comment above:`  // IF Method is OPTIONS we want to return after setting headers` ? Doesn't it tell us that we indeed can not return right now?

Reply:
We set headers above, like cors.